homeassistant/components/igloohome/__init__.py

```diff
@@ -0,0 +1,50 @@
+"""The igloohome integration."""
+
+from __future__ import annotations
+
+from aiohttp import ClientError
+from igloohome_api import (
+    Api as IgloohomeApi,
+    ApiException,
+    Auth as IgloohomeAuth,
+    AuthException,
+    GetDevicesResponse,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_CLIENT_ID, CONF_CLIENT_SECRET, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+type IgloohomeConfigEntry = ConfigEntry[(IgloohomeApi, GetDevicesResponse)]
```

Comment:
I'd rather prefer a `dataclass` for the content of the ConfigEntry (instead of a tuple)